subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -1099,7 +1103,8 @@ public Object property(String propertyName) throws MissingPropertyException {
 
     @Override
     public Object findProperty(String propertyName) {
-        return hasProperty(propertyName) ? property(propertyName) : null;
```

Comment:
Trying to guess the reason for this change. A passerby may have the same question. Maybe add a note on the "why"?

Reply:
It's in the commit message. It's a good idea to also add a comment, though. Thanks!

Reply:
Ah:

> Avoid duplicate problem reported on inherited property access within `findProperty`

Are the exact problem reported in the two cases? If so, and maybe that is not something we can do anything about now, but should be up to the thing collecting and reporting errors to ensure the same problem (kind, message, location) were never reported in duplication.

Reply:
The two problems reported from `hasProperty` and `getProperty` have the same message but different stacktraces, which normally means that they originate from different pieces of logic. Having only one of `hasProperty` or `getProperty` report the problem would miss calls to the other, which are done in one or another function in the project's public API. I doubt that it would be feasible to invent some heuristic clustering for this case unless it's a common issue across the codebase.

Reply:
👍